BitmarkSDK/Bitmark/EventSubscription.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  EventSubscription.swift
+//  BitmarkSDK
+//
+//  Created by Anh Nguyen on 6/13/19.
+//  Copyright © 2019 Bitmark. All rights reserved.
+//
+
+import Foundation
+import SwiftCentrifuge
+
+public struct BitmarkChangedInfo {
+    public let bitmarkId: String
+    public let txId: String
+    public let presence: Bool
+}
+
+extension BitmarkChangedInfo: Decodable {
+    enum CodingKeys: String, CodingKey
+    {
+        case bitmarkId = "bitmark_id"
+        case txId = "tx_id"
+        case presence = "presence"
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        bitmarkId = try values.decode(String.self, forKey: .bitmarkId)
+        txId = try values.decode(String.self, forKey: .txId)
+        presence = try values.decode(Bool.self, forKey: .presence)
+    }
+}
+
+
+
+public class EventSubscription {
+    public static let shared = EventSubscription()
+    internal var client: CentrifugeClient?
+    internal var account: String?
+    
+    public func connect(_ signable: KeypairSignable) throws {
+        let api = API()
+        let token = try api.requestWSToken(signable: signable)
+        
+        guard let endpointHost = api.endpoint.apiServerURL.host else {
```

Comment:
Hi a @anhnguyenbitmark , 
I'm not sure, but I'm curious that should we make this check host presence command before making requestWSToken above?
cause I see in requestWSToken, it's calling API and use apiServerURL.

Reply:
That's to prevent the case when we input a wrong apiServerURL to the endpoint, then the value of `api.endpoint.apiServerURL.host` will be `nil`.

Reply:
I mean in requestWSToken function,
you are calling: let requestURL = endpoint.apiServerURL.appendingPathComponent("/v3/ws-auth"); which requires `apiServerURL.host` also. 
so is it better to check host presence before making `let token = try api.requestWSToken(signable: signable)`?

Reply:
Ah, good catch! I'll update.